### ISSUE TITLE
Filter Auto-Update at every browser-restart, button not working, advanced tab with not functionality

### ISSUE THREAD

**Issue by Dispositiv** [gorhill/uBlock#3519]
### Describe the issue
While working with OSX / little-snitch and Linux / opensnitch and with ublock in the background, every time the browser starts the whole Auto-Update-procedure of ublock is started – and with a working network-filter/'snitch' you then see *lots* of connection-requests (to the filter-sites): this is totally annoying. every time the browser starts. *e*v*e*r*y* time. 

* yes, the button 'auto-update filter-list' under '3rd party' is 'off' - while clicking on it does not do anything.
* advanced user / auto-update-period resets always to '7' at every restart, setting it to something like '4096' does not do anything on the problem.

### Your settings

- OS/version: OSX 10.13.3 / Debian
- Browser/version: Firefox 59b10 - user.js according to https://github.com/ghacksuserjs/ghacks-user.js
- uBlock Origin version: 1.15.6

##### Your filter lists

"Default filter lists"

**Comment by gorhill** [comment]
You probably have custom browser settings preventing uBO from saving its resources, such that it is forced to download some resources (those not part of the package) at each restart.

**Comment by Dispositiv** [comment]
ok, i see - you saw user.js changes and thought that. i do not implement the whole package and do check on every change made to the user.js

the issue i elaborated is relatively new, since FF58 / last uBO-versions since November i *guess* - and no, like every other addon in my config, uBO should and does in principle not have any problems to save its resources. 

it simply does not see and hear any changes to the option 'auto-update filter-list' the user makes - while every other option is followed by another option 'apply changes' (and does survive a restart) this is not (and does not). again: all this is relatively new and for my understanding it seems to be connected in a change of code on your side.

**Comment by gorhill** [comment]
> yes, the button 'auto-update filter-list' under '3rd party' is 'off' - while clicking on it does not do anything.

You will need to investigate more on your side, it's un-reproducible on my side.

Try installing uBO in a new profile _without_ any changes to browser settings, and without any other extension installed.

> every time the browser starts the whole Auto-Update-procedure of ublock is started – and with a working network-filter/'snitch' you then see lots of connection-requests (to the filter-sites):

Provide information about what exactly is supposedly "updating" at browser start -- this is key information you should already have provided. uBO will pull nothing from remote servers with default filter lists, so you customized uBO and did not disclose this as demanded in [CONTRIBUTING](https://github.com/gorhill/uBlock/blob/master/CONTRIBUTING.md). Even if you manually added more filter lists which are **not** part of the package, these are **not** pulled from remote servers at launch time, so there can possibly be "lots of connection-requests" -- unless uBO is prevented from using indexedDB because of you custom settings.

> auto-update-period resets always to '7' at every restart,

Duplicate of #3387.

**Comment by Thorin-Oakenpants** [comment]
I use the ghacks user.js as is (minor overrides not relevant) and have zero issues with extension settings/files-downloads/local storage being saved for any extension. @Dispositiv , since you have uM working fine, can I assume you have allowed a cookie (all cookies are blocked by default) for your uBO UUID - see https://github.com/ghacksuserjs/ghacks-user.js/wiki/4.1.1-Setting-Extension-Permission-Exceptions

Other than that, do some troubleshooting ( https://github.com/ghacksuserjs/ghacks-user.js/wiki/1.4-Troubleshooting ) - step 1: can you reproduce without the user.js, in a nilla profile, etc. If you can determine the user.js is at fault, then let me know at our repo

**Comment by gorhill** [comment]
Closing, as OP is not coming back to provide needed information to make the case that there is an issue with uBO. uBO will connect to remote servers only when a needed resource is not available locally. Given that uBO caches remote resources, it will never connect to remote servers to fetch these resource unless an update operation is forced. uBO never auto-update at launch time, only after ~7 minutes, and only for resources which needs to be updated, and only if auto-update is enabled.

If uBO is unable to cache the resources it fetches, then the user needs to investigate what custom browser config or OS config causes this and address this on their side, this is outside of uBO's control.
